src/avatar/draw.js

```diff
@@ -28,58 +31,165 @@ createNameSpace("realityEditor.avatar.draw");
         }
     }
 
-    let renderedNames = null;
+    // show a list of circular icons, one per avatar, with the (random) color and (chosen) initials of that user
     function renderAvatarIconList(connectedAvatars) {
-        if (JSON.stringify(connectedAvatars) === renderedNames) {
-            return;
-        }
-
-        console.log('updated  names: ', connectedAvatars);
-        
         let iconContainer = document.getElementById('avatarIconContainer');
-        if (!iconContainer) {
-            iconContainer = document.createElement('div');
-            iconContainer.id = 'avatarIconContainer';
-            iconContainer.classList.add('avatarIconContainerScaleAdjustment')
-            iconContainer.style.position = 'absolute';
-            iconContainer.style.top = (realityEditor.device.environment.variables.screenTopOffset + 20) + 'px';
-            iconContainer.style.left = '50vw';
-            iconContainer.style.transform = 'translateX(-50%)';
-            document.body.appendChild(iconContainer)
-        }
-        
+        if (!iconContainer) { createIconContainer(); }
         while (iconContainer.hasChildNodes()) {
```

Comment:
@benptc  iconContainer will be null here after the first createIconContainer

Reply:
thanks for the catch!